### ISSUE TITLE
condition episodic memory on time

### ISSUE THREAD

**Issue by pieroit** [pieroit/cheshire-cat#10]
When the cat retrieves relevant episodic memories (things user said in the past), it is important to:
- sort them by time (most recent memories are more important)
- insert into the memory text info about time (e.g. "yesterday", "days ago", "years ago")

**Comment by salvog7** [comment]
Hi Piero, I would like to fix this issue (within the limit of my abilities), how can I do it?

**Comment by pieroit** [comment]
> Hi Piero, I would like to fix this issue (within the limit of my abilities), how can I do it?

[Here](https://github.com/pieroit/cheshire-cat/blob/main/core/cat/looking_glass/cheshire_cat.py#L132) episodic memories are extracted.

Each memory passed to the function should already contain both a timestamp and the text itself.
So, given the memory timestamp and the current timestamp:
 - calculate delta (now - timestamp)
 - convert delta to [datetime.timedelta](https://docs.python.org/3/library/datetime.html#timedelta-objects)
 - use something like [this](https://gist.github.com/sodle/222155df86117d5c4ebcc2e272c1d233) to convert the delta to a conversational format (i.e. "2 weeks ago")
 - append the human readable delta to the memory. So `I went to the bar` becomes `I went to the bar (2 weeks ago)`

Tell me if you encounter problems
